Create docker-compose.proper.yml

frozenfoxx opened the pull request:
Created a corresponding "proper" setup Docker Compose file. This will serve as a guide for a user to tweak the paths on to build the setup described in the "proper" section without building launcher scripts or similar.

frozenfoxx committed:
Create docker-compose.proper.yml

davestephens commented:
Thanks for this, a good addition! 

I think you're missing a few ports for a "proper" setup though, or maybe host networking would be better instead given it's reasonably trivial for a user to switch service ports?

frozenfoxx commented:
I definitely wouldn't mind having extra ports added, I just didn't see any on the README.md under the "proper" section, only 8888. My feeling is always that a real user is going to create a custom docker-compose.yaml based on one of the samples in the repo so it should only reflect what's available in the README.

We could update it to do 8086 as well for SSH to this and the README, though I'm not sure if Enigma requires any special options to terminate that by default.

davestephens commented:
What do you think about using host mode networking then it doesn't matter?

On Fri, 25 Jan 2019, 18:25 FrozenFOXX <notifications@github.com wrote:

> I definitely wouldn't mind having extra ports added, I just didn't see any
> on the README.md under the "proper" section, only 8888. My feeling is
> always that a real user is going to create a custom docker-compose.yaml
> based on one of the samples in the repo so it should only reflect what's
> available in the README.
>
> We could update it to do 8086 as well for SSH to this and the README,
> though I'm not sure if Enigma requires any special options to terminate
> that by default.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/davestephens/docker-enigma-bbs/pull/5#issuecomment-457671324>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAozMNt-I_HFtCvYOqGb__WKi8neXCzoks5vG0wXgaJpZM4aR7ii>
> .
>

frozenfoxx commented:
I don't have an issue with that at all and can definitely change it as such. We'd probably want to update the "proper" example in the README to reflect it as well, but that'd also give prospective users two different networking examples. I really like it when users get plenty of example configurations to work with so that would definitely help out. I'll go ahead and change it to host networking.

frozenfoxx committed:
Added network_mode: host

Updated to remove explicit ports and add network mode for the "proper" compose example.

frozenfoxx committed:
Update docker-compose.proper.yml

frozenfoxx commented:
I fixed the deploy options and verified this docker-compose file will function.
